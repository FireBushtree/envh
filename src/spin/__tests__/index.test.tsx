import React from 'react';
import { render } from '@testing-library/react';
import Spin from '../index';

const SPINNING_ELEMENT_CLASS = '.eh-spin.is-spinning';

describe('<Spin>', () => {
  test('render default should spinning', () => {
    const { container } = render(<Spin>spin</Spin>);
    expect(container).toMatchSnapshot();
    const spinEl = container.querySelector(SPINNING_ELEMENT_CLASS);
    expect(spinEl).toBeTruthy();
  });

  test('set spinning props change', () => {
    const { container, rerender } = render(<Spin spinning={false}>spin</Spin>);
    let spinEl = container.querySelector(SPINNING_ELEMENT_CLASS);
    expect(spinEl).toBe(null);

    rerender(<Spin spinning>spin</Spin>);
    spinEl = container.querySelector(SPINNING_ELEMENT_CLASS);
    expect(spinEl).toBeTruthy();
  });
});
