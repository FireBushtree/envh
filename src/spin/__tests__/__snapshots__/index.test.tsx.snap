// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spin> render default should spinning 1`] = `
<div>
  <div
    class="eh-spin is-spinning"
  >
    <div
      class="eh-spin-roller"
    >
      <div />
    </div>
    <div
      class="eh-spin-mask"
    />
    <div
      class="eh-spin-children"
    >
      spin
    </div>
  </div>
</div>
`;
